File: string/palindrome.c
#include<stdio.h>
main()
{
    int i,l,count=0,j;
    char c[20],d[20];
    printf("Enter the string:\n");
    gets(c);
    for(i=0;c[i]!=0;i++);
    printf("%d",i);
    for(j=0;j<i;j++)
    {
    for(l=i-1;l>=0;l--)
      {
        if(c[l]==c[j])
            count++;
        else
            break;
      }
    }
    if(count>0)
        printf("The string is a palindrome string.\n");
    else
        printf("The string is not a palindrome string.\n");
}

File: Python/palindrome.py
class PalindromeChecker:
    def __init__(self,value):
        self.value = value

    def is_Palindrome():
        raise NotImplemented("Subclass must implement the method!")
    
class StringPalindrome(PalindromeChecker):
    def is_Palindrome(self):
        cleaned_value = ''.join(self.value.lower().split())
        if cleaned_value == cleaned_value[::-1]:
            return True
        
class IntegerPalindrome(PalindromeChecker):
    def is_Palindrome(self):
        if str(self.value) == str(self.value)[::-1]:
            return True
        
def checkPalindrome(Palindrome_checker):
    if Palindrome_checker.is_Palindrome():
        print("It is a palindrome!")

    else:
        print("Not a palindrome!")

strchk = StringPalindrome(input("Enter a word: "))
intchk = IntegerPalindrome(int(input("Enter a number: ")))

checkPalindrome(strchk)
checkPalindrome(intchk)
        